src/_pytest/fixtures.py

```diff
@@ -1032,21 +1062,44 @@ def fixture(scope="function", params=None, autouse=False, ids=None, name=None):
                 ``fixture_<fixturename>`` and then use
                 ``@pytest.fixture(name='<fixturename>')``.
     """
-    if callable(scope) and params is None and autouse is False:
+    if isinstance(callable_or_scope, str):
+        warnings.warn(SCOPE_AS_NON_KEYWORD, stacklevel=2)
+        scope = callable_or_scope
+        callable_or_scope = None
+
+    if callable_or_scope and params is None and autouse is False:
         # direct decoration
-        return FixtureFunctionMarker("function", params, autouse, name=name)(scope)
+        return FixtureFunctionMarker("function", params, autouse, name=name)(
```

Comment:
I wonder, does it not make sense to pass `scope` here now, instead of `"function"`? Since now it is possible to do:

```py
pytest.fixture(myfunc, scope="session")
```

Reply:
Yes, that would make a perfect sense. I would say that it is actually a bug that the scope would be ignored. Thanks.